integrationtests/src/main/java/io/kroxylicious/proxy/testcluster/ClusterConfig.java

```diff
@@ -0,0 +1,205 @@
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Singular;
+import lombok.ToString;
+
+@Builder(toBuilder = true)
+@Getter
+@ToString
+public class ClusterConfig {
+
+    /**
+     * if true, cluster will use an in-VM kafka
+     */
+    private final Boolean inVM;
+    /**
+     * if true, the cluster will be brought up in Kraft-mode
+     */
+    private final Boolean kraftMode;
+
+    /**
+     * name of SASL mechanism to be configured on kafka for the external listener, if null, anonymous communication
+     * will be used.
+     */
+    private final String saslMechanism;
+    @Builder.Default
+    private Integer brokersNum = 1;
+
+    private final String kafkaKraftClusterId = Uuid.randomUuid().toString();
+    /**
+     * The users and passwords to be configured into the server's JAAS configuration used for the external listener.
+     */
+    @Singular
+    private final Map<String, String> users;
+
+    public Stream<ConfigHolder> getBrokerConfigs(Supplier<KafkaEndpoints> endPointConfigSupplier, Supplier<Endpoint> zookeeperEndpointSupplier) {
+        List<ConfigHolder> properties = new ArrayList<>();
+        KafkaEndpoints kafkaEndpoints = endPointConfigSupplier.get();
+        for (int brokerNum = 0; brokerNum < brokersNum; brokerNum++) {
+            Properties server = new Properties();
+
+            server.put("broker.id", Integer.toString(brokerNum));
+
+            var interBrokerEndpoint = kafkaEndpoints.getInterBrokerEndpoint(brokerNum);
+            var clientEndpoint = kafkaEndpoints.getClientEndpoint(brokerNum);
+
+            // - EXTERNAL: used for communications to/from consumers/producers
+            // - INTERNAL: used for inter-broker communications (always no auth)
+            // - CONTROLLER: used for inter-broker controller communications (kraft - always no auth)
+
+            var externalListenerTransport = saslMechanism == null ? "PLAINTEXT" : "SASL_PLAINTEXT";
+
+            var protocolMap = new TreeMap<>();
+            var listeners = new TreeMap<>();
+            var advertisedListeners = new TreeMap<>();
+            protocolMap.put("EXTERNAL", externalListenerTransport);
+            listeners.put("EXTERNAL", clientEndpoint.getBind().toString());
+            advertisedListeners.put("EXTERNAL", clientEndpoint.getConnect().toString());
+
+            protocolMap.put("INTERNAL", "PLAINTEXT");
+            listeners.put("INTERNAL", interBrokerEndpoint.getBind().toString());
+            advertisedListeners.put("INTERNAL", interBrokerEndpoint.getConnect().toString());
+            server.put("inter.broker.listener.name", "INTERNAL");
+
+            if (isKraftMode()) {
+                var controllerEndpoint = kafkaEndpoints.getControllerEndpoint(brokerNum);
+                var quorumVoters = IntStream.range(0, this.brokersNum)
+                        .mapToObj(b -> String.format("%d@%s", b, kafkaEndpoints.getControllerEndpoint(b).getConnect().toString())).collect(Collectors.joining(","));
```

Comment:
In Kraft mode, the `node.id` is also a required parameter. I guess it didn't restrict in v3.2. But it is required in v3.3 and later. ref: [here](https://kafka.apache.org/documentation/#brokerconfigs)

Reply:
Done